expression/builtin_time.go

```diff
@@ -2684,19 +2688,25 @@ func (du *baseDateArithmitical) getIntervalFromDecimal(ctx sessionctx.Context, a
 		/* keep interval as original decimal */
 	case "SECOND":
 		// Decimal's EvalString is like %f format.
+		neg = false
```

Comment:
It would be better not to do the extra revise code in these paths.
Also, would you like to add a unit test to cover it?